jaxrs-common/src/main/java/io/micronaut/jaxrs/common/JaxRsInterceptedWrite.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.jaxrs.common;
+
+import io.micronaut.context.BeanRegistration;
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.convert.ConversionService;
+import io.micronaut.core.type.Argument;
+import io.micronaut.core.type.MutableHeaders;
+import io.micronaut.http.MediaType;
+import io.micronaut.http.codec.CodecException;
+import jakarta.ws.rs.ext.WriterInterceptor;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * The JAX-RS body write interceptor.
+ *
+ * @param <T> The type
+ * @author Denis Stepanov
+ * @since 4.9.0
+ */
+@Internal
+public abstract class JaxRsInterceptedWrite<T> {
```

Comment:
can this be sealed, does it need to be public?

Reply:
Those are used as anonymous classes